test/system/src/test/java/io/pravega/test/system/StreamCutsTest.java

```diff
@@ -0,0 +1,279 @@
+        URI zkUri = zkUris.get(0);
+
+        // 2. Check if bk is running, otherwise start, get the zk ip.
+        Service bkService = Utils.createBookkeeperService(zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        // 3. Start controller.
+        Service conService = Utils.createPravegaControllerService(zkUri);
+        if (!conService.isRunning()) {
+            conService.start(true);
+        }
+
+        List<URI> conUris = conService.getServiceDetails();
+        log.debug("Pravega controller service details: {}", conUris);
+
+        // 4.Start segmentstore.
+        Service segService = Utils.createPravegaSegmentStoreService(zkUri, conUris.get(0));
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega segmentstore service details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+        streamManager = StreamManager.create(controllerURI);
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream one", streamManager.createStream(SCOPE, STREAM_ONE,
+                StreamConfiguration.builder().scope(SCOPE).streamName(STREAM_ONE)
+                                   .scalingPolicy(ScalingPolicy.fixed(PARALLELISM_ONE)).build()));
+        assertTrue("Creating stream two", streamManager.createStream(SCOPE, STREAM_TWO,
+                StreamConfiguration.builder().scope(SCOPE).streamName(STREAM_TWO)
+                                   .scalingPolicy(ScalingPolicy.fixed(PARALLELISM_TWO)).build()));
+    }
+
+    /**
+     * This test verifies the correct operation of readers using StreamCuts. Concretely, the test creates two streams
+     * with different number of segments and it writes some events (TOTAL_EVENTS) in them. Afterwards, the test creates
+     * a list of StreamCuts that encompasses both streams every CUT_SIZE events. The test asserts that new groups of
+     * readers can be initialized at these StreamCut intervals and that only CUT_SIZE events are read. Finally, this
+     * test also verifies that an existing reader group can be reset to the starting position in the stream.
+     */
+    @Test
+    public void streamCutsTest() {
+        @Cleanup
+        ClientFactory clientFactory = ClientFactory.withScope(SCOPE, controllerURI);
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, controllerURI);
+        readerGroupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().stream(Stream.of(SCOPE, STREAM_ONE))
```

Comment:
Move `.stream` in this line to the next.

Reply:
Done.